File: annotation-platform/backend/app/schemas/project.py
from pydantic import BaseModel
from typing import Optional
from uuid import UUID
from datetime import datetime


class ProjectCreate(BaseModel):
    name: str
    description: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class ProjectRead(BaseModel):
    id: UUID
    name: str
    description: Optional[str] = None
    status: str
    created_by: Optional[UUID] = None
    created_at: datetime
    updated_at: datetime
    image_count: int = 0
    annotated_count: int = 0
    approved_count: int = 0

    model_config = {"from_attributes": True}

File: annotation-platform/frontend/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  description?: string;
  status: "active" | "archived";
  created_by?: string;
  created_at: string;
  updated_at: string;
  image_count: number;
  annotated_count: number;
  approved_count: number;
}
